File: helidon-se/src/main/java/com/github/frederikpietzko/dto/CreatePetDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Pet;
import com.github.frederikpietzko.model.Type;

public record CreatePetDto(
  Long id,
  String name,
  int age,
  Type type) {

  public Pet toEntity() {
    return new Pet(id, name, age, type);
  }
}

File: ktor/src/main/kotlin/com/github/frederikpietzko/dto/CreatePetDto.kt
package com.github.frederikpietzko.dto

import com.github.frederikpietzko.model.Type
import kotlinx.serialization.Serializable

@Serializable
data class CreatePetDto(
  val id: Long?,
  val name: String,
  val age: Int,
  val type: Type,
) {
  fun toModel() = com.github.frederikpietzko.model.Pet(
    id = id,
    name = name,
    age = age,
    type = type,
  )
}

